src/Bundle/Bundle.php

```diff
@@ -0,0 +1,350 @@
+	 *
+	 * @param ProductRow $row
+	 */
+	public function addProductRow(ProductRow $row)
+	{
+		$key = md5(serialize([$row->getProductID(), $row->getOptions()]));
+
+		if (array_key_exists($key, $this->_productRows)) {
+			$this->_productRows[$key]->increaseQuantity($row->getQuantity());
+		} else {
+			$this->_productRows[$key] = $row;
+		}
+	}
+
+	/**
+	 * Remove all products from the bundle
+	 */
+	public function clearProducts()
+	{
+		$this->_productRows = [];
+	}
+
+	/**
+	 * Get all product rows from the bundle
+	 *
+	 * @return array
+	 */
+	public function getProductRows()
+	{
+		return $this->_productRows;
+	}
+
+	/**
+	 * Set the price for a specific currency
+	 *
+	 * @param int | float | string $price      The price to set
+	 * @param string $currencyID               The currency ID
+	 * @throws \InvalidArgumentException       Throws exception if $price is not numeric
+	 * @throws \InvalidArgumentException       Throws exception if $currencyID is not a string or is more than three
+	 *                                         characters
+	 */
+	public function setPrice($price, $currencyID)
+	{
+		if (!is_numeric($price)) {
+			throw new \InvalidArgumentException('Price must be numeric');
+		}
+
+		if (!is_string($currencyID) || !preg_match('/^[A-Za-z]+$/', $currencyID)) {
+			throw new \InvalidArgumentException('Currency ID must be a string of no more than three letters');
+		}
+
+		$this->_prices[strtoupper($currencyID)] = round($price, 2);
+	}
+
+	/**
+	 * Get the price for a specific currency
+	 *
+	 * @param string $currencyID               The currency ID corresponding to the price
+	 * @throws \InvalidArgumentException       Throws exception if $currencyID is not a string or is more than three
+	 *                                         characters
+	 * @throws \LogicException                 Throws exception if currency is not set in prices array
+	 *
+	 * @return mixed
+	 */
+	public function getPrice($currencyID = null)
+	{
+		$currencyID = $currencyID ?: $this->_defaultCurrency;
+
+		if (!is_string($currencyID) || !preg_match('/^[A-Za-z]+$/', $currencyID)) {
+			throw new \InvalidArgumentException('Currency ID must be a string of no more than three letters');
+		}
+
+		$currencyID = strtoupper($currencyID);
+
+		if (!array_key_exists($currencyID, $this->_prices)) {
+			throw new \LogicException('Currency with ID of `' . $currencyID . '` not set on bundle');
+		}
+
+		return $this->_prices[$currencyID];
+	}
+
+	public function getPrices()
+	{
+		return $this->_prices;
+	}
+
+	/**
+	 * Set whether the bundle can be used in conjunction with discount codes
+	 *
+	 * @param bool $allowCodes     Set to true to allow bundle to work with discount codes
+	 */
+	public function setAllowCodes($allowCodes)
+	{
+		$this->_allowCodes = (bool) $allowCodes;
+	}
+
+	/**
+	 * Returns true if bundle allows codes. If `setAllowCodes()` has not be run, it will return false by default
+	 *
+	 * @return bool
+	 */
+	public function allowCodes()
```

Comment:
This function is misleading. It sounds like it sets allowCodes to true. perhaps "allowsCodes"?